### PR TITLE
Resource mapper properly encodes special characters in the uri

### PR DESCRIPTION
**Pull request by jaxoncreed**
Fix for #1324

### CODE REVIEW

File: lib/resource-mapper.js [hunk 1]
```diff
@@ -78,7 +78,10 @@ class ResourceMapper {
 
     // Determine the URL by chopping off everything after the dollar sign
     const pathname = this._removeDollarExtension(path)
-    const url = `${this.resolveUrl(hostname)}${encodeURI(pathname)}`
+    console.log(pathname)
```
Comment (RubenVerborgh) [inline lib/resource-mapper.js]: ping

File: lib/resource-mapper.js [hunk 2]
```diff
@@ -78,7 +78,10 @@ class ResourceMapper {
...
+    const url = `${this.resolveUrl(hostname)}${
+      pathname.split('/').map((component) => encodeURIComponent(component)).join('/')
+    }`
```
Comment (RubenVerborgh) [inline lib/resource-mapper.js]: I would split on the filesystem path separator char [`path.sep`](https://nodejs.org/docs/latest/api/path.html#path_path_sep).

Comment (jaxoncreed) [reply lib/resource-mapper.js]: Gotcha

**Comment by jaxoncreed** [comment]
Continued in https://github.com/solid/node-solid-server/pull/1337

**Comment by RubenVerborgh** [comment]
Changed base of this PR instead.

**Pull request merged**